src/Illuminate/Mail/Message.php

```diff
@@ -319,11 +330,32 @@ public function attachData($data, $name, array $options = [])
     /**
      * Embed a file in the message and get the CID.
      *
-     * @param  string  $file
+     * @param  string|\Illuminate\Contracts\Mail\Attachable|\Illuminate\Mail\Attachment  $file
      * @return string
      */
     public function embed($file)
     {
+        if ($file instanceof Attachable) {
+            $file = $file->toMailAttachment();
+        }
+
+        if ($file instanceof Attachment) {
+            return $file->attachWith(
+                function ($path) use ($file) {
+                    $cid = $file->as ?? Str::random();
```

Comment:
Maybe use `random(10)` like the below.

```suggestion
                    $cid = $file->as ?? Str::random(10);
```

Is there a huge reason why you prefer `$file->as` and not just always pick `Str::random(10)`? Could simplify things a bit.

Reply:
This is the name that is given to the file / image / document in the email. If you drag the image / document from the email onto your own system you get the nicer name specified by the developer, rather than an extension-less file...

Felt it was nicer and more consistent with attachments as well. I think we should allow the `embed` function to accept a name - but as that would be breaking I didn't want to make that change, of course.

![Screen Shot 2022-06-03 at 5 26 00 pm](https://user-images.githubusercontent.com/24803032/171807993-a574ab6c-23ef-43d6-8f5b-e6680d3a74c0.png)

Reply:
but not a hill I'm gonna die on. Happy for it to just always give a random name if we prefer 👍

Reply:
oh, and for "web based" mail clients, they often show a list of attachments - inline or not, and it will show the name there as well.

So mostly just that it isn't completely hidden and the name is sometimes exposed to people.

Reply:
Definitely makes sense to use the file name I think. All good 👍